pl_bolts/datamodules/emnist_datamodule.py

```diff
@@ -0,0 +1,221 @@
+from typing import Any, Callable, Optional, Union
+
+from pl_bolts.datamodules.vision_datamodule import VisionDataModule
+from pl_bolts.transforms.dataset_normalizations import emnist_normalization
+from pl_bolts.utils import _TORCHVISION_AVAILABLE
+from pl_bolts.utils.warnings import warn_missing_pkg
+
+if _TORCHVISION_AVAILABLE:
+    from torchvision import transforms as transform_lib
+    from torchvision.datasets import EMNIST
+else:  # pragma: no cover
+    warn_missing_pkg('torchvision')
+    EMNIST = object
+
+
+class EMNISTDataModule(VisionDataModule):
+    """
+    .. figure:: https://user-images.githubusercontent.com/4632336/123210742-4d6b3380-d477-11eb-80da-3e9a74a18a07.png
+        :width: 400
+        :alt: EMNIST
+
+    .. list-table:: Dataset information (source: `EMNIST: an extension of MNIST to handwritten
+        letters <https://arxiv.org/abs/1702.05373>`_ [Table-II])
+       :header-rows: 1
+
+       * - Split Name
+         - No. classes
+         - Train set size
+         - Test set size
+         - Validation set
+         - Total size
+       * - ``"byclass"``
+         - 62
+         - 697,932
+         - 116,323
+         - No
+         - 814,255
+       * - ``"byclass"``
+         - 62
+         - 697,932
+         - 116,323
+         - No
+         - 814,255
+       * - ``"bymerge"``
+         - 47
+         - 697,932
+         - 116,323
+         - No
+         - 814,255
+       * - ``"balanced"``
+         - 47
+         - 112,800
+         - 18,800
+         - Yes
+         - 131,600
+       * - ``"digits"``
+         - 10
+         - 240,000
+         - 40,000
+         - Yes
+         - 280,000
+       * - ``"letters"``
+         - 37
+         - 88,800
+         - 14,800
+         - Yes
+         - 103,600
+       * - ``"mnist"``
+         - 10
+         - 60,000
+         - 10,000
+         - Yes
+         - 70,000
+
+    |
+
+    Here is the default EMNIST, train, val, test-splits and transforms.
+
+    Transforms::
+
+        emnist_transforms = transform_lib.Compose([
+            transform_lib.ToTensor(),
+        ])
+
+    Example::
+
+        from pl_bolts.datamodules import EMNISTDataModule
+        dm = EMNISTDataModule('.')
+        model = LitModel()
+        Trainer().fit(model, datamodule=dm)
+    """
+    name = "emnist"
+    dataset_cls = EMNIST
+    dims = (1, 28, 28)
+
+    _official_val_split = {
+        'balanced': 18_800,
+        'digits': 40_000,
+        'letters': 14_800,
+        'mnist': 10_000,
+    }
+
+    def __init__(
+        self,
+        data_dir: Optional[str] = None,
+        split: str = 'mnist',
+        val_split: Union[int, float] = 0.2,
+        num_workers: int = 16,
+        normalize: bool = False,
+        batch_size: int = 32,
+        seed: int = 42,
+        shuffle: bool = False,
+        pin_memory: bool = False,
```

Comment:
```suggestion
        shuffle: bool = True,
        pin_memory: bool = True,
```

Reply:
@tchaton These are the same defaults used in other existing datamodules.

```python
num_workers: int = 16,
shuffle: bool = False,
pin_memory: bool = False,
```

Some of the existing datamodules where the above defaults are used currently.

- [x] `binary_mnist_datamodule`
- [x] `cifar10_datamodule`
- [x] `mnist_datamodule`
- [x] `cityscapes_datamodule`
- [x] `fashion_mnist_datamodule`
- [x] `imagenet_datamodule`

So, in the light of this, I would suggest that the same defaults, for at least `shuffle` and `pin_memory`, be used for `emnist_datamodule` and `binary_emnist_datamodule`. If any changes must be made, let that be part of a separate issue and make the changes to all of them in a separate PR.

cc: @akihironitta @Borda

Reply:
Opened a quick PR #701 for this suggestion, so resolving this thread.